### ISSUE TITLE
Default values for parameters of arrays of components should be displayed with the correct dimensions

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#11847]
Consider the following MWE:
```modelica
model Test
  import Modelica.Constants.pi;
  Modelica.Electrical.Analog.Sources.SineVoltage sineVoltage[3] annotation(
    Placement(transformation(origin = {10, 10}, extent = {{-10, -10}, {10, 10}}, rotation = -90)));
equation
  annotation(
    uses(Modelica(version = "4.0.0")),
    Diagram(coordinateSystem(extent = {{-100, 100}, {180, -80}})),
    experiment(StartTime = 0, StopTime = 0.1, Tolerance = 1e-06, Interval = 0.0002));
end Test;
```
If you double-click on the sineVoltage component array, you see this:
![immagine](https://github.com/OpenModelica/OpenModelica/assets/4233148/afb8eeab-cf77-4e4b-a98e-5f34f309bd44)

There are two problems here that should be fixed:
- [ ] there is no way to tell that the parameters that need to be input are for an array of size three, so it is easy to make mistakes with their dimensions, or to forget to set `each`. My suggestion is that one should see name: sineVoltage[3]
- [ ] the default values are shown as scalars, while they should be shown as arrays, either as `{0, 0, 0}` or `fill(0, 3)`. I would prefer the first version if the number of dimensions is small (say, <= 4, which works for 3-phase, 3D mechanics and quaternions), the second version for larger arrays, to avoid overbloating it.

Keeping @ceraolo in the loop

**Comment by casella** [comment]
@adeas31 this should be fixed with high priority in 1.26.0

**Comment by casella** [comment]
@adeas31 maybe we can still sneak this in 1.25.1?

**Comment by adeas31** [comment]
I suggest not to include this in 1.25.1. This was neither a bug nor a regression. This is a new code and functionality which should be part of 1.26.

**Comment by casella** [comment]
Well, the shown scalar parameters are definitely a bug IMHO. And a quite confusing one: if you actually provide the same value you see as default, you get an error.

BTW, after #11710, shouldn't we see the array dimensions in that parameter dialog?

**Comment by adeas31** [comment]
> Well, the shown scalar parameters are definitely a bug IMHO. And a quite confusing one: if you actually provide the same value you see as default, you get an error.

Not really. The shown default value is the value at the definition of the variable. When you provide the same value it is added as a binding of the array variable so it becomes wrong. All other Modelica tools are also showing the default values as scalar. So that is an enhancement in OpenModelica that we improvised how to display the default value of array variables.

> BTW, after https://github.com/OpenModelica/OpenModelica/issues/11710, shouldn't we see the array dimensions in that parameter dialog?

No, that only added the array dimensions to the parameters listed in the parameter dialog. With #13948 we now also display array dimensions with the name of the component for whom the parameter dialog is opened.

**Comment by adeas31** [comment]
I ported this to maintenance/v1.25 branch with #14011. It was needed to make fix of #14002 work properly.
